### PR TITLE
Remove QuantizationConfig fields patch from NodeActivationQuantizationConfig

### PR DESCRIPTION
**Pull request by irenaby**
## Pull Request Description:
Remove the intermediate patch that injected fields from QuantizationConfig back to NodeActivationQuantizationConfig.
Use QuantizationConfig directly in all places that read it from the node candidates.
Remove min_threshold from TrainableQuantizerActivationConfig.

## Checklist before requesting a review:
- [ ] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [ ] I have added new unittest (if necessary).

### CODE REVIEW

File: tests/keras_tests/feature_networks_tests/feature_networks/network_editor/edit_error_method_test.py [hunk 1]
```diff
@@ -1,70 +0,0 @@
...
-    ChangeCandidatesActivationQuantConfigAttr
-from model_compression_toolkit.core.common.network_editors.node_filters import NodeTypeFilter
-from tests.keras_tests.feature_networks_tests.base_keras_feature_test import BaseKerasFeatureNetworkTest
-from tests.keras_tests.utils import get_layers_from_model_by_type
-
-keras = tf.keras
-layers = keras.layers
-
-
-class EditActivationErrorMethod(BaseKerasFeatureNetworkTest):
```
Comment (yarden-yagil-sony) [inline tests/keras_tests/feature_networks_tests/feature_networks/network_editor/edit_error_method_test.py:36]: Why this test is deleted? and if it's due to changes, don't we want to create a new version of it with adaptations?

File: model_compression_toolkit/trainable_infrastructure/common/trainable_quantizer_config.py [hunk 2]
```diff
@@ -44,7 +44,6 @@ def __init__(self,
                  activation_n_bits: int,
                  activation_quantization_params: Dict,
                  enable_activation_quantization: bool,
-                 min_threshold: float,
```
Comment (yarden-yagil-sony) [inline model_compression_toolkit/trainable_infrastructure/common/trainable_quantizer_config.py:47]: why is it removed from the quantizer config?

Comment (irenaby) [reply model_compression_toolkit/trainable_infrastructure/common/trainable_quantizer_config.py:47]: There is also a MIN_THRESHOLD const and in some places the const was used directly and in some places it came from the config (which has the same default value and most probably no one ever used a non-default value). So it was decided to just consistently use the const and remove it from the config.

File: tests/keras_tests/feature_networks_tests/feature_networks/network_editor/edit_error_method_test.py (hunk 1 above)
Comment (irenaby) [reply tests/keras_tests/feature_networks_tests/feature_networks/network_editor/edit_error_method_test.py:36]: The whole edit rule was removed as it edited a field that is no more part of the node activation config, so there is nothing to replace with
